xdsl/dialects/onnx.py

```diff
@@ -932,6 +932,86 @@ def verify_(self) -> None:
             )
 
 
+@irdl_op_definition
+class Transpose(IRDLOperation):
+    """
+    The transpose_tensor function takes a tensor as input and returns its transpose.
+    Transposing a tensor means flipping its dimensions, so that rows become columns and vice versa.
+    """
+
+    name = "onnx.Transpose"
+
+    T = Annotated[AnyFloat | IntegerType, ConstraintVar("T")]
+    tensor_input = operand_def(TensorType[T])
+
+    perm = opt_attr_def(ArrayAttr[AnyIntegerAttr], attr_name="perm")
+
+    tensor_output = result_def(TensorType[T])
+
+    assembly_format = (
+        "`(` $tensor_input `)` attr-dict `:` `(` type($tensor_input) "
+        "`)` `->` type($tensor_output) "
+    )
+
+    def __init__(self, tensor_input: SSAValue, perm: Attribute):
+        super().__init__(
+            attributes={"perm": perm},
+            operands=[tensor_input],
+            result_types=[tensor_input.type],
+        )
+
+    def verify_(self) -> None:
+        if not isinstance(
+            tensor_input_type := self.tensor_input.type, TensorType
+        ) or not isinstance(tensor_output_type := self.tensor_output.type, TensorType):
+            assert (
+                False
+            ), "onnx elementwise operation operands and result must be of type TensorType"
+
+        tensor_input_shape = tensor_input_type.get_shape()
+        tensor_output_shape = tensor_output_type.get_shape()
+
+        # numbers in perm cannot be repeated
+        if self.perm is not None:
```

Comment:
Since all the checks below rely on self.perm not being None, we can flip this:
```suggestion
        if self.perm is None:
            return
```